src/System.Net.Security/src/System/Net/Unix/CertModule.cs

```diff
@@ -140,8 +143,46 @@ internal override string[] GetRequestCertificateAuthorities(SafeDeleteContext se
 
         internal override X509Store EnsureStoreOpened(bool isMachineStore)
         {
-            // TODO (Issue #3362) do the implementation
-            return null;
+            if (isMachineStore)
+            {
+                // There's not currently a LocalMachine\My store on Unix, so don't bother trying
+                // and having to deal with the exception.
+                return null;
+            }
+
+            return EnsureStoreOpened(ref s_userCertStore, StoreLocation.CurrentUser);
+        }
+
+        private static X509Store EnsureStoreOpened(ref X509Store storeField, StoreLocation storeLocation)
+        {
+            if (storeField == null)
+            {
+                lock (s_lockObject)
+                {
+                    if (storeField == null)
+                    {
+                        try
+                        {
+                            X509Store store = new X509Store(StoreName.My, storeLocation);
+                            store.Open(OpenFlags.ReadOnly);
+
+                            storeField = store;
```

Comment:
Well, to keep the handy ref-based helper apparently I'd need to throw a MemoryBarrier() call in right before the field assignment.  If we didn't bring MemoryBarrier into CoreFX; I guess I'll have to un-helper and slap on the volatile keyword.